### PR TITLE
Bug 1845232 - Adds action to clear private browsing

### PR DESCRIPTION
**Pull request by boek**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1845232

### CODE REVIEW

**Review by HarrisonOg — APPROVED** [review]
Looks good!

**Comment by DonalMe** [comment]
@boek this has test failures

File: fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/interactor/BrowserToolbarInteractor.kt [hunk 1]
```diff
@@ -27,6 +27,11 @@ interface BrowserToolbarInteractor {
      * Navigates to the Home screen. Called when a user taps on the Home button.
      */
     fun onHomeButtonClicked()
+
+    /**
+     * Deletase all tabs and navigates to the Home screen. Called when a user taps on the erase button.
```
Comment (gabrielluong) [inline fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/interactor/BrowserToolbarInteractor.kt:32]:
```suggestion
     * Deletes all tabs and navigates to the Home screen. Called when a user taps on the erase button.
```

File: android-components/components/ui/icons/src/main/res/drawable/mozac_ic_data_clearance.xml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- This Source Code Form is subject to the terms of the Mozilla Public
+   - License, v. 2.0. If a copy of the MPL was not distributed with this
+   - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="25dp"
+    android:height="24dp"
+    android:viewportWidth="25"
```
Comment (gabrielluong) [inline android-components/components/ui/icons/src/main/res/drawable/mozac_ic_data_clearance.xml:8]:
```suggestion
    android:width="24dp"
    android:height="24dp"
    android:viewportWidth="24"
```

**Pull request merged**